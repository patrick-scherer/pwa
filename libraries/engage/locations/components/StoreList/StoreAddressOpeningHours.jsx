import React from 'react';
import PropTypes from 'prop-types';
import { getWeekDaysOrder } from '@shopgate/engage/core';
import StoreAddressOpeningHoursLine from './StoreAddressOpeningHoursLine';
import { openingHours } from './style';

/**
 * Renders the store's opening hours.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function StoreAddressOpeningHours({ hours }) {
  if (!hours) {
    return null;
  }

  return (
    <div className={openingHours}>
      {getWeekDaysOrder().map(weekDay => (
        <StoreAddressOpeningHoursLine hours={hours[weekDay]} code={weekDay} key={weekDay} />
      ))}
    </div>
  );
}

StoreAddressOpeningHours.propTypes = {
  hours: PropTypes.shape(),
};

StoreAddressOpeningHours.defaultProps = {
  hours: null,
};

export default StoreAddressOpeningHours;
