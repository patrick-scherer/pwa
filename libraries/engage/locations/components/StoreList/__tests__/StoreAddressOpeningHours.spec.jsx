import React from 'react';
import { shallow, mount } from 'enzyme';
import StoreAddressOpeningHours from '../StoreAddressOpeningHours';

jest.mock('@shopgate/engage/core');

const hours = {
  mon: '8:00am - 4:00pm',
  tue: '',
  wed: '8:00am - 8:00pm',
};

describe('<StoreAddressOpeningHours />', () => {
  it('should not render if no hours are passed', () => {
    const wrapper = shallow(<StoreAddressOpeningHours />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance()).toEqual(null);
  });

  it('should not render hours that are empty', () => {
    const wrapper = mount(<StoreAddressOpeningHours hours={hours} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('[data-test-id="hours-mon"]').text()).toEqual(hours.mon);
    expect(wrapper.find('[data-test-id="hours-tue"]').getElements().length).toEqual(0);
    expect(wrapper.find('[data-test-id="hours-wed"]').text()).toEqual(hours.wed);
  });
});
