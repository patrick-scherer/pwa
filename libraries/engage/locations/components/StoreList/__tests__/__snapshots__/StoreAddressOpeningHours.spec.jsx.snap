// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreAddressOpeningHours /> should not render hours that are empty 1`] = `
<StoreAddressOpeningHours
  hours={
    Object {
      "mon": "8:00am - 4:00pm",
      "tue": "",
      "wed": "8:00am - 8:00pm",
    }
  }
>
  <div
    className="css-1xm8nj0"
  >
    <StoreAddressOpeningHoursLine
      code="mon"
      hours="8:00am - 4:00pm"
      key="mon"
    >
      <div
        className="css-9shxm7"
      >
        <div
          className="css-1ppc7ry"
        >
          locations.mon
        </div>
        <div
          data-test-id="hours-mon"
        >
          8:00am - 4:00pm
        </div>
      </div>
    </StoreAddressOpeningHoursLine>
    <StoreAddressOpeningHoursLine
      code="tue"
      hours=""
      key="tue"
    />
    <StoreAddressOpeningHoursLine
      code="wed"
      hours="8:00am - 8:00pm"
      key="wed"
    >
      <div
        className="css-9shxm7"
      >
        <div
          className="css-1ppc7ry"
        >
          locations.wed
        </div>
        <div
          data-test-id="hours-wed"
        >
          8:00am - 8:00pm
        </div>
      </div>
    </StoreAddressOpeningHoursLine>
    <StoreAddressOpeningHoursLine
      code="thu"
      hours={null}
      key="thu"
    />
    <StoreAddressOpeningHoursLine
      code="fri"
      hours={null}
      key="fri"
    />
    <StoreAddressOpeningHoursLine
      code="sat"
      hours={null}
      key="sat"
    />
    <StoreAddressOpeningHoursLine
      code="sun"
      hours={null}
      key="sun"
    />
  </div>
</StoreAddressOpeningHours>
`;

exports[`<StoreAddressOpeningHours /> should not render if no hours are passed 1`] = `""`;
